homeassistant/components/sky_hub/device_tracker.py

```diff
@@ -31,72 +37,134 @@ class SkyHubDeviceScanner(DeviceScanner):
 
     def __init__(self, config):
         """Initialise the scanner."""
-        _LOGGER.info("Initialising Sky Hub")
+        self._connection_failed = False
+        self._dataparse_failed = False
+        self._log_message("Initialising Sky Hub", level=_INFO)
```

Comment:
We don't want side effects like logging in init method. Please remove this.

Reply:
I've not changed the structure of this other than to change to async and remove the exceptions. I'm happy to leave it to the original developer to re-write if needed. If I have time in the future I'll come back and look at it again.

Reply:
@MartinHjelmare - I have now submitted an updated version with devices access code broken out.